### DISCUSSION TITLE
ZigBeeBridge with light function and IR receiver

### DISCUSSION THREAD

**Discussion by Micha70** [arendst/Tasmota#17298] [Q&A] [answered]
Hello,
I'm trying to use Zigbeebridge  (SONOFF ZB Bridge) also to control a RGB light stribe with IR receiver. The light stribe works as expected. Build a small driver on external PCB. However the IR receiver I don't get working. First the IRrecv I could not configure for the input. I'm using GPIO5. Figured out I need to create my own custom build. So made following modification in tasmota_configurations.h:

```
#ifdef FIRMWARE_ZBBRIDGE
...
//#undef USE_IR_REMOTE                 // Disable IR driver
//06.12.2022 additional enabled
#define USE_IR_REMOTE                   // Send IR remote commands using library IRremoteESP8266 and ArduinoJson (+4k code, 0k3 mem, 48 iram)
  #define USE_IR_RECEIVE                // Support for IR receiver (+5k5 code, 264 iram)
//06.12.2022 additional enabled

```

Build was done and target updated sucessfully.

Now I was able to configure GPIO5 with IRrecv. But still not working. In console window I dont't get any information about pressed button on remote control like {"IrReceived":{"Protocol":"<value>","Bits":<value>,"Data":<value>}}

I have similar setup with a small RGB controller and standard tasmota build and is working with the IR receiver.

What I'm missing? Can anybody give me a short hint. 

Regards
Michael

**Comment by s-hadinger** [comment]
If the `IRecv` appears in the template then there is no more configuration needed. It's most certainly a hardware issue. How did you check that the receiver works?

**Reply by Micha70** [reply]
I was checking with an Oszilloscope on pin24 of the ESP which should be GPIO5. But I see now the voltage is at about 3.3V if no button is pressed and when pressing goes only down to 2.8V during the pulses. This is maybe not ok ....

**Reply by s-hadinger** [reply]
Definitely a hardware problem. Nothing we can help with unfortunately

**Reply by Micha70** [reply]
I'm not sure if there is something else is connected on pin24. It looks like is unconnected. Is there any schematic available for the SONOFF ZB Bridge?

**Reply by Micha70** [reply]
if I discinnect the receiver in receiver output the level is correct .... 0...3.3V

**Comment by Micha70** [comment, marked as answer]
works now. I have used GPIO14 and on PIN9. There is a pad on bottom to connect (SCLK)
